### PR TITLE
feat(component): Add component page

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

Issue: #10 

### Description: 

Feature Add Component page

### Suggest Reviewer: 

@hoangnt2  @heliocastro

### CODE REVIEW

File: src/components/AddAdditionalRoles.tsx [hunk 1]
```diff
@@ -47,29 +62,41 @@ export default function AddAdditionalRolesComponent() {
...
+                                        <>
+                                            <option value="Stakeholder">Stakeholder</option>
+                                            <option value="Analyst">Analyst</option>
+                                            <option value="Contributor">Contributor</option>
+                                            <option value="Accountant">Accountant</option>
+                                            <option value="End User">End User</option>
+                                            <option value="Quality Manager">Quality Manager</option>
+                                            <option value="Test Manager">Test Manager</option>
+                                            <option value="Technical writer">Technical writer</option>
+                                            <option value="Key User">Key User</option>
```
Comment (heliocastro) [inline src/components/AddAdditionalRoles.tsx:84]: Please translate all strings

File: src/components/AddAdditionalRoles.tsx [hunk 2]
```diff
@@ -47,29 +62,41 @@ export default function AddAdditionalRolesComponent() {
...
+                                        </>
+                                        }
                                     </select>
                                 </div>
                                 <div className="col-lg-5">
                                     <input name="email" value={elem.email} type="email" onChange={e => handleInputChange(e, j)} className="form-control" placeholder={`Enter email`} aria-describedby={`Email`} />
                                 </div>
                                 <div className="col-lg-2">
-                                    <button type="button" onClick={() => handleRemoveClick(j)} className={`fw-bold btn btn-light ${styles['button-plain']}`}><i className="bi bi-trash3-fill"></i></button>
+                                    <button type="button" onClick={() => handleRemoveClick(j)} className={`fw-bold btn btn-light ${styles['button-plain']}`}><FaTrashAlt className="bi bi-trash3-fill"></FaTrashAlt></button>
```
Comment (heliocastro) [inline src/components/AddAdditionalRoles.tsx:93]: Don'tt add inline styles, use global css if the button is common, or component css for specialized

File: src/components/ComponentAddSummary/ComponentAddSummary.tsx
Comment (heliocastro) [inline src/components/ComponentAddSummary/ComponentAddSummary.tsx:1]: Separate the logic from the component ui
Please break up in more components as well

Comment (heliocastro) [reply src/components/ComponentAddSummary/ComponentAddSummary.tsx:1]: Remember to translate all strings

File: src/components/SearchUsersModal.tsx
Comment (heliocastro) [inline src/components/SearchUsersModal.tsx:1]: Place component under the sw360 namespace

File: src/components/SearchModerators/SelectTableModerators.tsx
Comment (heliocastro) [inline src/components/SearchModerators/SelectTableModerators.tsx:1]: Place component under sw360 namespace

File: src/components/SearchModerators/ModeratorsTable.tsx
Comment (heliocastro) [inline src/components/SearchModerators/ModeratorsTable.tsx:1]: Place the component under sw360 namespace

File: src/components/SearchModerators/ModeratorsDiaglog.tsx
Comment (heliocastro) [inline src/components/SearchModerators/ModeratorsDiaglog.tsx:1]: Place component under sw360 namespace

File: src/components/SearchComponentOwner/SelectTableComponentOwner.tsx
Comment (heliocastro) [inline src/components/SearchComponentOwner/SelectTableComponentOwner.tsx:1]: Place component under sw360 namespace

File: src/components/SearchComponentOwner/ComponentOwnerDialog.tsx
Comment (heliocastro) [inline src/components/SearchComponentOwner/ComponentOwnerDialog.tsx:1]: Place component under sw360 namespace

File: src/components/SelectCountry.tsx
Comment (heliocastro) [inline src/components/SelectCountry.tsx:1]: Put component under sw360 namespace

**Comment by tuannn2** [comment]
Thank you for your reviewing, @heliocastro 
I have updated. Please check it!

File: src/components/ComponentAddSummary/ComponentAddSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        const roles = {
+            Contributor: contributors,
+            Committer: commiters,
+            Expert: expecters,
+        }
+        return roles
+    }
+
+    const notify = (text: string, type: TypeOptions) =>
+        toast(text, {
```
Comment (hoangnt2) [inline src/components/ComponentAddSummary/ComponentAddSummary.tsx:116]: Translate notify's message

File: src/components/sw360/SearchModerators/ModeratorsDiaglog.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            const data = await response.json()
+            return data
+        } else {
+            notFound()
+        }
+    }, [])
+
+    useEffect(() => {
+        fetchData(`users`).then((users: any) => {
+            console.log(users)
```
Comment (hoangnt2) [inline src/components/sw360/SearchModerators/ModeratorsDiaglog.tsx:60]: remove log

File: src/components/AddAdditionalRoles.tsx [hunk 5]
```diff
@@ -33,44 +45,62 @@ export default function AddAdditionalRolesComponent() {
...
+                                        </>
+                                        }
                                     </select>
                                 </div>
                                 <div className="col-lg-5">
                                     <input name="email" value={elem.email} type="email" onChange={e => handleInputChange(e, j)} className="form-control" placeholder={`Enter email`} aria-describedby={`Email`} />
                                 </div>
                                 <div className="col-lg-2">
-                                    <button type="button" onClick={() => handleRemoveClick(j)} className={`fw-bold btn btn-light ${styles['button-plain']}`}><i className="bi bi-trash3-fill"></i></button>
+                                    <button type="button" onClick={() => handleRemoveClick(j)} className={`fw-bold btn btn-light button-plain`}><FaTrashAlt className="bi bi-trash3-fill"></FaTrashAlt></button>
```
Comment (hoangnt2) [inline src/components/AddAdditionalRoles.tsx:96]: replace by
```
 <FaTrashAlt className="bi bi-trash3-fill" />
```

File: src/components/AddKeyValue.tsx [hunk 6]
```diff
@@ -8,67 +8,105 @@
...
+                                    aria-describedby={`${props.keyName.toLowerCase()} value`}
+                                />
+                            </div>
+                            <div className='col-lg-2'>
+                                <button
+                                    type='button'
+                                    onClick={() => handleRemoveClick(j)}
+                                    className={`fw-bold btn btn-light button-plain`}
+                                >
+                                    <FaTrashAlt className='bi bi-trash3-fill'></FaTrashAlt>
```
Comment (hoangnt2) [inline src/components/AddKeyValue.tsx:93]: replace by 
```
<FaTrashAlt className="bi bi-trash3-fill">
```

**Comment by tuannn2** [comment]
Thank you for your reviewing, @hoangnt2 
I have updated.

**Review by heliocastro — APPROVED** [review]
Good to go

**Pull request merged**